[UnifiedPDF] [iOS] WKWebView._displayingPDF is wrong when the feature is enabled

aprotyas opened the pull request:
#### daf9470519bad4b9aec8e24ba6121732aab675e1
<pre>
[UnifiedPDF] [iOS] WKWebView._displayingPDF is wrong when the feature is enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=280311">https://bugs.webkit.org/show_bug.cgi?id=280311</a>
<a href="https://rdar.apple.com/136634661">rdar://136634661</a>

Reviewed by Wenson Hsieh.

When UnifiedPDFEnabled==YES, we do not have the WKPDFView custom content
view on the iOS family. However, currently, regardless of feature
enablement, we only check for the presence of an appropriate custom
content view for the PDF mime type to determine if we are presenting a
PDF document on the iOS family.

This patch widens the scope of the &quot;PDF document on the main frame&quot;
check currently inquired for macOS only, being careful to fallback to
existing checks in case UnifiedPDFEnabled==NO.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isDisplayingPDF]):

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(TestWebKitAPI::UNIFIED_PDF_TEST):

Add an API test exercising WKWebView._displayingPDF with Unified PDF
enabled. Test coverage for when the feature is disabled is covered by
the WKWebView.IsDisplayingPDF test.

Canonical link: <a href="https://commits.webkit.org/284206@main">https://commits.webkit.org/284206@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bf6810ba4b83942ea5f37779c3903afd5a970091

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68744 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48136 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21403 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/72814 "Hash bf6810ba for PR 34203 does not build (failure)") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19889 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/70861 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55932 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/19705 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/72814 "Hash bf6810ba for PR 34203 does not build (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13222 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71811 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43978 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59354 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/72814 "Hash bf6810ba for PR 34203 does not build (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40644 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/16780 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18247 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62593 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17128 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/74507 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12715 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16373 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/62274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12755 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59436 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62308 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10269 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3880 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10478 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43937 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45011 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46205 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44753 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
```diff
@@ -167,6 +169,20 @@
+// FIXME: Combine this test with the WKWebView.IsDisplayingPDF test.
+UNIFIED_PDF_TEST(WKWebView_IsDisplayingPDF)
+{
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 600, 600) configuration:configurationForWebViewTestingUnifiedPDF().get() addToWindow:YES]);
```
Nit - I think we prefer explicit `RetainPtr` these days.

aprotyas replied:
Good point, fixed!

aprotyas committed:
[UnifiedPDF] [iOS] WKWebView._displayingPDF is wrong when the feature is enabled
https://bugs.webkit.org/show_bug.cgi?id=280311
rdar://136634661

Reviewed by Wenson Hsieh.

When UnifiedPDFEnabled==YES, we do not have the WKPDFView custom content
view on the iOS family. However, currently, regardless of feature
enablement, we only check for the presence of an appropriate custom
content view for the PDF mime type to determine if we are presenting a
PDF document on the iOS family.

This patch widens the scope of the "PDF document on the main frame"
check currently inquired for macOS only, being careful to fallback to
existing checks in case UnifiedPDFEnabled==NO.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isDisplayingPDF]):

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(TestWebKitAPI::UNIFIED_PDF_TEST):

Add an API test exercising WKWebView._displayingPDF with Unified PDF
enabled. Test coverage for when the feature is disabled is covered by
the WKWebView.IsDisplayingPDF test.

Canonical link: https://commits.webkit.org/284206@main
